apps/webapp/app/models/runtimeEnvironment.server.ts

```diff
@@ -7,6 +7,9 @@ export async function findEnvironmentByApiKey(apiKey: string) {
   const environment = await prisma.runtimeEnvironment.findUnique({
     where: {
       apiKey,
+      project: {
```

Comment:
This will block API calls if the project is deleted

Reply:
I wouldn't add this to the query, but just check the return value. This could introduce an inefficient query in a hot path so easier to just check afterwards and return undefined.

Reply:
Good idea